packages/@react-spectrum/list/src/ListView.tsx

```diff
@@ -173,6 +173,8 @@ function ListView<T extends object>(props: SpectrumListProps<T>, ref: DOMRef<HTM
           }
 
           // TODO: Best way to implement only for when closest can be dropped on
+          // TODO: Figure out the typescript for this
+          // @ts-ignore
           if (y >= r.y + 10 && y <= r.maxY - 10 && collection.getItem(closest.key).value.type === 'folder') {
```

Comment:
ts-ignoring this for now because `value.type` doesn't fit the generic type that `value` gets here. Will update the logic when we decide whether or not to make the type more restrictive for ListView (aka user has to provide type to their items) or if we change this logic overall

Reply:
Turns out `value.type` is not alway going to be defined, but only specifically for static ListViews which I don't think makes sense w/ drag and drop in general. Perhaps just to be safer we should change this to be `collection.getItem(closest.key).hasChildNodes` but we should discuss what the true restrictions for "on" drops should be. Will handle in followup